### DISCUSSION TITLE
feature request: preference to automatically verify local data upon completion of torrent

### DISCUSSION THREAD

**Discussion by mortonsanchez** [transmission/transmission#3577] [Ideas]
I'm sorry if this is the wrong place for this (or if this feature has already been requested). If it is the wrong place, please let me know where the correct place for this would be. 

Currently using Transmssion for Mac 3.00+ (9afbf58c4e)

I would like to see a preference/setting to enable Transmission to automatically verify local data upon completion of a torrent. 

Sometimes when I open a file that has "finished", it doesn't open or play correctly. I then go to Transmission, choose "Verify Local Data", and more often than not, the torrent is incomplete.

The incompleteness may or may not be Transmission's fault (usually it's because my computer crashed, or I had to force quit Transmission). Honestly, I don't care where the fault lies. But qBittorrent has this feature (to auto-verify after completion), and hopefully it is not very difficult one to implement.

Thanks!

**Comment by i3laze** [comment]
+1 here.
App knows download became corrupt and does nothing about it.

Actually, triggering a manual verify from smartphone is a real issue:
https://github.com/transmission/transmission/issues/464

**Comment by jeremyelgin** [comment]
I would like this too. I have noticed at times a lot of torrents will not have completed properly (even though nothing noticeable has gone wrong - no crashes, nothing at all... just a download that completes but a manual verify determines that something/more needs to be re/downloaded), so now I manually check everything, and this is where I notice issues, sometimes a lot, sometimes none (which I think causes people to assume there is no fault when there is) and sometimes just randomly once-in-a-blue-moon (like today). Please, save us this extra step.
Yes, qBittorrent (which I liked for a while due to it's RSS scrapping but do not use now because I believe it has memory holes, and is so much more complicated) does this, it could be an easy fix to add a checkbox in the Settings and perform (call) the verify function upon "completion."

**Comment by abdullah1116** [comment]
+1

**Comment by Shiroe93** [comment]
+1

**Comment by rndusr** [comment]
The daemon has a `torrent_complete_verify_enabled` setting which does exactly this. It is documented here: https://github.com/transmission/transmission/blob/main/docs/Editing-Configuration-Files.md

**Reply by Shiroe93** [reply]
Not exactly 
Verify on start and on finish are not verify on error 
Still useful though

**Reply by rndusr** [reply]
The feature request is "to automatically verify local data upon completion of torrent". Where do you get the error condition from?

Transmission (and every other client) should always hash check every downloaded piece and re-download it the piece is faulty. But that doesn't help if the piece is stored on a faulty HDD or the file is altered after the piece is stored.

**Reply by Shiroe93** [reply]
You are right i misread the request i assumed it was on error because sometimes i found torrent that become corrupted after sometime and i have to start a check manually 
I hopped it was possible to start a check when an error is detected (transmission does infact tell me data are corrupted but does nothing)

**Comment by bdfortin** [comment]
In addition, for transfers with multiple files can there be a setting to verify individual files as each one completes?
